CMakeLists.txt

```diff
@@ -61,16 +61,15 @@ endif ()
 
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES ".*Clang") # both Clang and AppleClang
   set (is_clang TRUE)
-  # TODO enable this version check
   if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" AND
-         CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0)
-    message (FATAL_ERROR "This project requires clang 5 or later")
+         CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+    message (FATAL_ERROR "This project requires clang 7 or later")
   endif ()
   # TODO min AppleClang version check ?
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
   set (is_gcc TRUE)
-  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
-    message (FATAL_ERROR "This project requires GCC 7 or later")
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.4)
+    message (FATAL_ERROR "This project requires GCC 7.4 or later")
```

Comment:
I think we can remove this check. The ICE with gcc 7.3 appears  to be a jenkins only issue.